fission-core/library/Fission/App/Types.hs

```diff
@@ -0,0 +1,64 @@
+module Fission.App.Types
+  ( module Fission.App.Name
+  , App(..)
+  ) where
+
+import           Data.Swagger     hiding (URL, url)
+
+import           Fission.Prelude
+
+import           Fission.App.Name
+import           Fission.URL
+
+
+data App = App
+  { urls       :: [URL]
+  , insertedAt :: UTCTime
+  , modifiedAt :: UTCTime
+  }
+  -- What about deriving Json of normal data?
+  -- deriving (Eq, Generic, Show, ToJSON)
+  deriving (Eq, Show)
+
+
+
+instance FromJSON App where
+  parseJSON = withObject "App" \obj -> do
+    urls       <- obj .: "urls"
```

Comment:
Pretty sure this is not accounting for serializing a `[URL]` type from json but went through a few blogs and couldn't find a `Decode.list obj "urls"` sort of thing, curious what the Aeson equivalent is?

Reply:
Aeson has a polymorphic `instance (FromJSON a) => FromJSON [a]` ([source file](https://www.stackage.org/haddock/lts-17.10/aeson-1.5.6.0/src/Data.Aeson.Types.FromJSON.html)), so it'll deserialize lists automatically with no intervention required.